eslintPlugin/src/noAnimatedStyleToNonAnimatedComponent.ts

```diff
@@ -0,0 +1,155 @@
+import { AST_NODE_TYPES, TSESLint, TSESTree } from '@typescript-eslint/utils';
+
+const rule: TSESLint.RuleModule<'animatedStyle' | 'sharedValue'> = {
+  create: function (context) {
+    return {
+      JSXOpeningElement(node: TSESTree.JSXOpeningElement) {
+        if (node.name.type === AST_NODE_TYPES.JSXMemberExpression) return; // Property-like namespace syntax <Animated.View>
+        if (node.name.type === AST_NODE_TYPES.JSXNamespacedName) return; // XML-based namespace syntax: <Animated:View>
+
+        const sourceCode = context.getSourceCode();
+        const tokensBefore = sourceCode.getTokensBefore(node);
+        const componentName = node?.name?.name;
+
+        function isVariableDefinedAs(
+          variableName: string,
+          expectedToken: string
+        ) {
+          let isAnimated = false;
+          const variableNameTokenIds: Array<number> = [];
+
+          tokensBefore.forEach((token, idx) => {
+            if (token.value === variableName) {
+              variableNameTokenIds.push(idx);
+            }
+          });
+
+          variableNameTokenIds.forEach((idx) => {
+            if (tokensBefore[idx + 2].value === expectedToken) {
```

Comment:
this seems to be a magical +2

Reply:
I've added a comment.

Reply:
I like the comment a lot, but it seems to be misleading, since `idx+2` is `=` there. I assume that `a` is actually `idx` there. You should make that table to refer to the actual situation here - meaning that `const` should be `idx - 1`.

Reply:
Indeed I was wondering whether to start with -1 or 0